diff --git a/Makefile b/Makefile
index 3a5d431..2e7855f 100644
--- a/Makefile
+++ b/Makefile
@@ -195,8 +195,8 @@ SUBARCH := $(shell uname -m | sed -e s/i.86/x86/ -e s/x86_64/x86/ \
 # "make" in the configured kernel build directory always uses that.
 # Default value for CROSS_COMPILE is not to prefix executables
 # Note: Some architectures assign CROSS_COMPILE in their arch/*/Makefile
-ARCH		?= $(SUBARCH)
-CROSS_COMPILE	?= $(CONFIG_CROSS_COMPILE:"%"=%)
+ARCH		?= arm
+CROSS_COMPILE	?= arm-fsl-linux-gnueabi-
 
 # Architecture as present in compile.h
 UTS_MACHINE 	:= $(ARCH)
diff --git a/arch/arm/boot/dts/imx6ul-14x14-evk.dts b/arch/arm/boot/dts/imx6ul-14x14-evk.dts
index c1623c9..f6b8b83 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-evk.dts
+++ b/arch/arm/boot/dts/imx6ul-14x14-evk.dts
@@ -59,6 +59,17 @@
 			enable-active-high;
 		};
 
+		reg_usb_vmmc: regulator@2 {
+			compatible = "regulator-fixed";
+			regulator-name = "vbus_otg";
+			pinctrl-0 = <&pinctrl_usb_otg1>;
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio1 2 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+		};
+
+
 		reg_gpio_dvfs: regulator-gpio {
 			compatible = "regulator-gpio";
 			pinctrl-names = "default";
@@ -112,11 +123,11 @@
 		compatible = "spi-gpio";
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_spi4>;
-		pinctrl-assert-gpios = <&gpio5 8 GPIO_ACTIVE_LOW>;
+		pinctrl-assert-gpios = <&gpio1 21 GPIO_ACTIVE_LOW>;
 		status = "okay";
 		gpio-sck = <&gpio5 11 0>;
 		gpio-mosi = <&gpio5 10 0>;
-		cs-gpios = <&gpio5 7 0>;
+		cs-gpios = <&gpio1 20 0>;
 		num-chipselects = <1>;
 		#address-cells = <1>;
 		#size-cells = <0>;
@@ -279,28 +290,29 @@
 	status = "okay";
 };
 
-&tsc {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_tsc>;
-	status = "okay";
-	xnur-gpio = <&gpio1 3 0>;
-	measure_delay_time = <0xffff>;
-	pre_charge_time = <0xfff>;
-};
-
 &uart1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart1>;
 	status = "okay";
 };
 
+/*
 &uart2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart2>;
-	fsl,uart-has-rtscts;
-	/* for DTE mode, add below change */
-	/* fsl,dte-mode; */
-	/* pinctrl-0 = <&pinctrl_uart2dte>; */
+	status = "okay";
+};
+*/
+
+&uart3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart3>;
+	status = "okay";
+};
+
+&uart4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart4>;
 	status = "okay";
 };
 
@@ -311,15 +323,14 @@
 
 &usbotg2 {
 	dr_mode = "host";
+	vbus-supply=<&reg_usb_vmmc>;
 	disable-over-current;
 	status = "okay";
 };
 
 &usdhc1 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc1>;
-	pinctrl-1 = <&pinctrl_usdhc1_100mhz>;
-	pinctrl-2 = <&pinctrl_usdhc1_200mhz>;
 	cd-gpios = <&gpio1 19 0>;
 	keep-power-in-suspend;
 	enable-sdio-wakeup;
@@ -333,30 +344,10 @@
 	no-1-8-v;
 	non-removable;
 	keep-power-in-suspend;
-	enable-sdio-wakeup;
+	bus-width = <8>;
 	status = "okay";
 };
 
-&i2c1 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c1>;
-	status = "okay";
-
-	mag3110@0e {
-		compatible = "fsl,mag3110";
-		reg = <0x0e>;
-		position = <2>;
-	};
-
-	fxls8471@1e {
-		compatible = "fsl,fxls8471";
-		reg = <0x1e>;
-		position = <0>;
-		interrupt-parent = <&gpio5>;
-		interrupts = <0 8>;
-	};
-};
 
 &i2c2 {
 	clock_frequency = <100000>;
@@ -372,6 +363,21 @@
 		wlf,shared-lrclk;
 	};
 
+	sii902x: sii902x@39 {
+	     compatible = "SiI,sii902x";
+	     pinctrl-names = "default";
+	     pinctrl-0 = <&pinctrl_sii902x>;
+	     interrupt-parent = <&gpio1>;
+	     interrupts = <18 IRQ_TYPE_EDGE_FALLING>;
+	     mode_str ="1280x720M@60";
+	     bits-per-pixel = <16>;
+	     rst-gpio0 = <&gpio_spi 0 0>;
+	     rst-gpio7 = <&gpio_spi 7 0>;
+	     reg = <0x39>;
+	     status = "okay";
+	 };
+
+
 	ov5640: ov5640@3c {
 		compatible = "ovti,ov5640";
 		reg = <0x3c>;
@@ -423,6 +429,20 @@
 			>;
 		};
 
+		pinctrl_usb_otg1: usbotg1grp {
+		      fsl,pins = <
+				MX6UL_PAD_GPIO1_IO02__GPIO1_IO02	0x17059
+			  >;
+		};
+
+		pinctrl_sii902x: hdmigrp-1 {
+		     fsl,pins = <
+				MX6UL_PAD_UART1_CTS_B__GPIO1_IO18	0x17059
+			 >;
+		 };
+
+
+
 		pinctrl_csi1: csi1grp {
 			fsl,pins = <
 				MX6UL_PAD_CSI_MCLK__CSI_MCLK		0x1b088
@@ -550,19 +570,24 @@
 			fsl,pins = <
 				MX6UL_PAD_UART2_TX_DATA__UART2_DCE_TX	0x1b0b1
 				MX6UL_PAD_UART2_RX_DATA__UART2_DCE_RX	0x1b0b1
-				MX6UL_PAD_UART3_RX_DATA__UART2_DCE_RTS	0x1b0b1
-				MX6UL_PAD_UART3_TX_DATA__UART2_DCE_CTS	0x1b0b1
 			>;
 		};
 
-		pinctrl_uart2dte: uart2dtegrp {
+		pinctrl_uart3: uart3grp {
 			fsl,pins = <
-				MX6UL_PAD_UART2_TX_DATA__UART2_DTE_RX	0x1b0b1
-				MX6UL_PAD_UART2_RX_DATA__UART2_DTE_TX	0x1b0b1
-				MX6UL_PAD_UART3_RX_DATA__UART2_DTE_CTS	0x1b0b1
-				MX6UL_PAD_UART3_TX_DATA__UART2_DTE_RTS	0x1b0b1
+				MX6UL_PAD_UART3_TX_DATA__UART3_DCE_TX 0x1b0b1
+				MX6UL_PAD_UART3_RX_DATA__UART3_DCE_RX 0x1b0b1
 			>;
 		};
+		
+		pinctrl_uart4: uart4grp {
+			fsl,pins = <
+				MX6UL_PAD_UART4_TX_DATA__UART4_DCE_TX 0x1b0b1
+				MX6UL_PAD_UART4_RX_DATA__UART4_DCE_RX 0x1b0b1
+			>;
+		};
+
+
 
 		pinctrl_usdhc1: usdhc1grp {
 			fsl,pins = <
@@ -605,24 +630,13 @@
 				MX6UL_PAD_NAND_DATA01__USDHC2_DATA1 0x17059
 				MX6UL_PAD_NAND_DATA02__USDHC2_DATA2 0x17059
 				MX6UL_PAD_NAND_DATA03__USDHC2_DATA3 0x17059
+				MX6UL_PAD_NAND_DATA04__USDHC2_DATA4 0x17059
+				MX6UL_PAD_NAND_DATA05__USDHC2_DATA5 0x17059
+				MX6UL_PAD_NAND_DATA06__USDHC2_DATA6 0x17059
+				MX6UL_PAD_NAND_DATA07__USDHC2_DATA7 0x17059
 			>;
 		};
 
-		pinctrl_tsc: tscgrp {
-			fsl,pins = <
-				MX6UL_PAD_GPIO1_IO01__GPIO1_IO01	0xb0
-				MX6UL_PAD_GPIO1_IO02__GPIO1_IO02	0xb0
-				MX6UL_PAD_GPIO1_IO03__GPIO1_IO03	0xb0
-				MX6UL_PAD_GPIO1_IO04__GPIO1_IO04	0xb0
-			>;
-		};
-
-		pinctrl_i2c1: i2c1grp {
-			fsl,pins = <
-				MX6UL_PAD_UART4_TX_DATA__I2C1_SCL 0x4001b8b0
-				MX6UL_PAD_UART4_RX_DATA__I2C1_SDA 0x4001b8b0
-			>;
-		};
 
 		pinctrl_i2c2: i2c2grp {
 			fsl,pins = <
@@ -659,6 +673,8 @@
 				MX6UL_PAD_BOOT_MODE1__GPIO5_IO11	0x70a1
 				MX6UL_PAD_SNVS_TAMPER7__GPIO5_IO07	0x70a1
 				MX6UL_PAD_SNVS_TAMPER8__GPIO5_IO08	0x80000000
+				MX6UL_PAD_UART2_TX_DATA__GPIO1_IO20	0x70a1
+				MX6UL_PAD_UART2_RX_DATA__GPIO1_IO21	0x70a1
 			>;
 		};
 
